projects/components/src/confirmation/confirmation-modal.component.ts

```diff
@@ -32,15 +32,18 @@ export class ConfirmationModalComponent {
   public readonly confirmButtonLabel: string;
   public readonly cancelButtonLabel: string;
   public readonly confirmButtonRole: ButtonRole;
-  public readonly descriptionText: string;
-  public readonly renderer?: TemplateRef<unknown>;
+  public readonly descriptionText!: string;
+  public readonly content!: TemplateRef<unknown>;
+  public readonly isContentString: boolean;
 
   public constructor(private readonly modalRef: ModalRef<boolean>, @Inject(MODAL_DATA) config: ConfirmationModalData) {
     this.confirmButtonLabel = config.confirmButtonLabel ?? ConfirmationModalComponent.DEFAULT_CONFIRM_LABEL;
     this.confirmButtonRole = config.confirmButtonRole ?? ConfirmationModalComponent.DEFAULT_CONFIRM_ROLE;
     this.cancelButtonLabel = config.cancelButtonLabel ?? ConfirmationModalComponent.DEFAULT_CANCEL_LABEL;
-    this.descriptionText = config.descriptionText ?? '';
-    this.renderer = config.content;
+    this.isContentString = isString(config.content);
+    this.isContentString
```

Comment:
nit: try to avoid assignments as expressions - can we move this to a traditional if/else?

Reply:
done. @aaron-steinfeld